.github/scripts/gerrit_query.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+
+# SPDX-License-Identifier: Apache-2.0
+# SPDX-FileCopyrightText: 2025 The Linux Foundation
+
+# Usage: ./gerrit_query.sh <gerrit_url> <patchset_number> <ssh_user_name>
+
+if [ "$#" -lt 3 ]; then
+  echo "Usage: $0 <gerrit_url> <patchset_number> <ssh_user_name>"
+  exit 1
+fi
+
+gerrit_url="$1"
+cid_patch_set_no="$2"
+ssh_user_name="$3"
+
+extract_project() {
+  echo "$1" | sed -E 's#.*/c/([^/]+/[^/]+)/\+.*#\1#'
+}
+
+extract_change_number() {
+  echo "$1" | sed -E 's#.*/\+/([0-9]+).*#\1#'
+}
+
+extract_hostname() {
+  echo "$1" | sed -E 's#https?://([^/]+)/.*#\1#'
+}
+
+gerrit_hostname=$(extract_hostname "$gerrit_url")
+change_number=$(extract_change_number "$gerrit_url")
+project=$(extract_project "$gerrit_url")
+
+if [ -n "$cid_patch_set_no" ]; then
+    ssh_command="ssh -p 29418 $ssh_user_name@$gerrit_hostname 'gerrit query --format=JSON project:$project change:$change_number owner:self is:open limit:1' --patch-sets"
+else
+    ssh_command="ssh -p 29418 $ssh_user_name@$gerrit_hostname 'gerrit query --format=JSON project:$project change:$change_number owner:self is:open limit:1' --current-patch-set"
+fi
```

Comment:
`ssh_command` should be set to the base before the if statement and _then_ extended with the appropriate `--patch-sets` or `--current-patch-set` options instead of repeating the full command twice.